Building FreeOrion 0.5 against Boost 1.85 fails

Doing a bulk build testing against Boost 1.85 we found that FreeOrion fails to build.

```
/tmp/pobj/freeorion-0.5/src-tarball/util/Directories.cpp:7:10: fatal error: 'boost/filesystem/convenience.hpp' file not found
#include <boost/filesystem/convenience.hpp>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

This appears to be due to this commit..

https://github.com/boostorg/filesystem/commit/5df060e95ca844fe91b29001b4ae22bdb65635c6

--- a/util/Directories.cpp
+++ b/util/Directories.cpp
@@ -4,7 +4,6 @@
 #include "i18n.h"
 
 #include <boost/algorithm/string/trim.hpp>
-#include <boost/filesystem/convenience.hpp>
 #include <boost/filesystem/operations.hpp>
 #include <boost/date_time/posix_time/posix_time.hpp>
 

